rtloader/common/stringutils.c

```diff
@@ -42,23 +42,33 @@ char *as_string(PyObject *object)
         // PyString_AsString might raise an error when python can't encode a
         // unicode string to byte
         PyErr_Clear();
-        return  NULL;
+        return NULL;
     }
     retval = strdupe(tmp);
 #else
-    if (!PyUnicode_Check(object)) {
-        return NULL;
-    }
-
-    PyObject *temp_bytes = PyUnicode_AsEncodedString(object, "UTF-8", "strict");
-    if (temp_bytes == NULL) {
-        // PyUnicode_AsEncodedString might raise an error if the codec raised an
-        // exception
-        PyErr_Clear();
-        return NULL;
+    PyObject *temp_bytes;
+
+    if (PyBytes_Check(object)) {
+        // We already have an encoded string, we suppose it has the correct encoding (UTF-8)
+        temp_bytes = object;
```

Comment:
I think you have to increase the ref count on `temp_bytes` here, otherwise the ref count decrement in `done` will be wrong (you'll be decrementing the ref count on the `PyObject` even though it was never increased).